### PR TITLE
Handle reflayers

### PR DESCRIPTION
**Pull request by staktrace**
I tested this to not regress behaviour, but I haven't tested that RefLayers actually get rendered properly. We might need to tweak this later but it should be fine to merge now.

### CODE REVIEW

File: gfx/layers/wr/WebrenderContainerLayer.cpp [hunk 1]
```diff
@@ -28,5 +28,16 @@ WebRenderContainerLayer::RenderLayer(wrstate* aWRState)
   wr_pop_dl_builder(aWRState, toWrRect(relBounds), toWrRect(relBounds), &transform.components[0], FrameMetrics::NULL_SCROLL_ID);
 }
 
+void
+WebRenderRefLayer::RenderLayer(wrstate* aWRState)
+{
+  WRScrollFrameStackingContextGenerator scrollFrames(aWRState, this);
+
+  gfx::Rect relBounds = TransformedVisibleBoundsRelativeToParent();
+  gfx::Matrix4x4 transform;// = GetTransform();
```
Comment (kvark) [inline gfx/layers/wr/WebrenderContainerLayer.cpp:37]: Does it have a sensible default constructor?

File: gfx/webrender/src/bindings.rs [hunk 2]
```diff
@@ -286,7 +286,7 @@ pub extern fn wr_create(width: u32, height: u32, counter: u32) -> *mut WrState {
     // let notifier = Box::new(Notifier::new(window.create_window_proxy()));
     // renderer.set_render_notifier(notifier);
 
-    let pipeline_id = PipelineId(0, counter);
+    let pipeline_id = PipelineId((layers_id >> 32) as u32, layers_id as u32);
```
Comment (kvark) [inline gfx/webrender/src/bindings.rs:289]: any reason you are not storing and passing `layers_id` as `PipelineId` right away? Seems to be converted like this in a few places.

File: gfx/layers/wr/WebrenderContainerLayer.cpp (hunk 1 above)
Comment (staktrace) [reply gfx/layers/wr/WebrenderContainerLayer.cpp:37]: Yes, it initializes to the identity matrix. I just copied this from the other Webrender*Layer classes which do the same thing.

File: gfx/webrender/src/bindings.rs (hunk 2 above)
Comment (staktrace) [reply gfx/webrender/src/bindings.rs:289]: I'm not sure what you mean by this. We have to do the conversion somewhere, whether it's in C++ or Rust code. And if it's in C++ then we have to create a C++ struct that maps to a PipelineId along with a function to do the conversion which seems like strictly more work.

Comment (kvark) [reply gfx/webrender/src/bindings.rs:289]: Ah, got it. All good then :)

**Pull request merged**